src/lib/data/dbDogs.ts

```diff
@@ -66,3 +66,58 @@ export async function dbSelectDog(
   }
   return null;
 }
+
+/**
+ * Inserts a dog_vet_preferences record
+ */
+export async function dbInsertDogVetPreference(
+  db: DbContext,
+  dogId: string,
+  vetId: string,
+): Promise<boolean> {
+  const sql = `
+  WITH mIdentifiers AS (
+    SELECT
+      user_id,
+      dog_id,
+      CAST($2 as BIGINT) as vet_id
+    FROM dogs
+    WHERE dog_id = $1
+    AND user_id IS NOT NULL
+  )
+
+  INSERT INTO dog_vet_preferences (user_id, dog_id, vet_id)
+  SELECT user_id, dog_id, vet_id FROM mIdentifiers
+  ON CONFLICT (dog_id, vet_id) DO NOTHING
+  RETURNING preference_creation_time
+  `;
+  const res = await dbQuery(db, sql, [dogId, vetId]);
+  return res.rows.length === 1;
+}
+
+export async function dbDeleteDogVetPreferences(
+  db: DbContext,
+  dogId: string,
+): Promise<number> {
+  const sql = `
+  WITH
+  mDeletion AS (
+    DELETE FROM dog_vet_preferences WHERE dog_id = $1
+    RETURNING *
```

Comment:
Can we use `RETURNING 1`? So it is clearer the actual values are not significant.